File: accessLayer/src/test/java/com/zeyad/usecases/utils/TestRealmModel.java
package com.zeyad.usecases.utils;

import android.support.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

import io.realm.RealmModel;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.RealmModule;

/**
 * @author by ZIaDo on 2/13/17.
 */
@RealmModule
public class TestRealmModel implements RealmModel {
    @SerializedName("id")
    @PrimaryKey
    private int id;
    @SerializedName("value")
    private String value;

    public TestRealmModel(int id, String value) {
        this.id = id;
        this.value = value;
        if (id <= 0)
            throw new IllegalArgumentException("id should be greater than 0");
    }

    public TestRealmModel() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getValue() {
        return value;
    }

    public void setValue(String value) {
        this.value = value;
    }

    @NonNull
    @Override
    public String toString() {
        return "TestRealmObject{" +
                "id=" + id +
                ", value='" + value + '\'' +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof TestRealmObject)) return false;
        TestRealmObject testRealmObject = (TestRealmObject) o;
        return getId() == testRealmObject.getId() && getValue().equals(testRealmObject.getValue());
    }

    @Override
    public int hashCode() {
        int result = getId();
        result = 31 * result + getValue().hashCode();
        return result;
    }
}

File: usecases/src/test/java/com/zeyad/usecases/TestRealmModel.kt
package com.zeyad.usecases

import android.arch.persistence.room.PrimaryKey
import com.google.gson.annotations.SerializedName

/**
 * @author by ZIaDo on 2/13/17.
 */
class TestRealmModel {
    @SerializedName("id")
    @PrimaryKey
    var id: Int = 0

    @SerializedName("value")
    var value: String? = null

    constructor(id: Int, value: String) {
        this.id = id
        this.value = value
        if (id <= 0) throw IllegalArgumentException("id should be greater than 0")
    }

    constructor()

    override fun toString(): String {
        return "TestRealmModel{" + "id=" + id + ", value='" + value + '\''.toString() + '}'.toString()
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o !is TestRealmModel) return false
        val testRealmObject = o as TestRealmModel?
        return id == testRealmObject!!.id && value == testRealmObject.value
    }

    override fun hashCode(): Int {
        var result = id
        result = 31 * result + value!!.hashCode()
        return result
    }
}
